lib/companies_house/client.rb

```diff
@@ -49,6 +50,14 @@ def officers(id)
       items
     end
 
+    def company_search(query, limit: 5)
+      request(
+        :company_search,
+        "search/companies",
+        q: query, items_per_page: limit,
```

Comment:
What happens if we don't send a limit? Could we not send one at all if one isn't specified? That way, we can make sure we get the API's default behaviour, rather than end up hard-coding a limit.

Reply:
I think this makes a lot of sense. On that note, I've changed this to `per_page` and added `offset` as well, both nullable, allowing us to properly paginate. The naming of those has been chosen to be consistent with the GoCardless naming convention.